File: backend/server/models/message.go
package models

import (
	"time"
)

type Conversation struct {
	ID        uint
	BuyerId   string
	SellerId  string
	PostId    string
	CreatedAt time.Time

	Buyer  *User
	Seller *User
	Post   *Post
}

type Message struct {
	ID             uint
	ConversationId string
	SenderId       string
	Content        string
	CreatedAt      time.Time

	Conversation *Conversation
	Sender       *User
}

File: frontend/ReSell/app/src/main/java/model/Message.kt
package model

import com.squareup.moshi.JsonClass
import java.time.LocalDateTime

@JsonClass(generateAdapter = true)
data class Message(
    val id: String,
    val conversationId: String,
    val senderId: String,
    val content: String,
    val createdAt: LocalDateTime?,
)
